internal/actions/update.go

```diff
@@ -51,128 +67,122 @@ func Update(client container.Client, params types.UpdateParams) (types.Report, e
 
 		if err != nil {
 			log.Infof("Unable to update container %q: %v. Proceeding to next.", targetContainer.Name(), err)
-			stale = false
-			staleCheckFailed++
-			progress.AddSkipped(targetContainer, err)
+			us.progress.AddSkipped(targetContainer, err)
+			containers[i].SetMarkedForUpdate(false)
 		} else {
-			progress.AddScanned(targetContainer, newestImage)
-		}
-		containers[i].SetStale(stale)
-
-		if stale {
-			staleCount++
+			us.progress.AddScanned(targetContainer, newestImage)
+			containers[i].SetMarkedForUpdate(shouldUpdate)
 		}
 	}
 
 	containers, err = sorter.SortByDependencies(containers)
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("failed to sort containers for updating: %v", err)
 	}
 
 	UpdateImplicitRestart(containers)
 
 	var containersToUpdate []types.Container
 	for _, c := range containers {
-		if !c.IsMonitorOnly(params) {
+		if c.ToRestart() {
```

Comment:
In testing my latest changes that assume this PR will be coming in, I found that I needed to change this:

		if c.ToRestart() {

to

		if c.ToRestart() && !c.IsMonitorOnly(params) {

In order to avoid the monitor only test cases failing. Consider whether to make the same change here in your code or just to incorporate the  	!c.IsMonitorOnly(params)  test within the definition of ToRestart()

Reply:
It's not needed anymore, since `shouldUpdate` (and later `markedForUpdate`) will never be true for `monitorOnly`.